packages/schema/lib/schemas/FieldMetaSchema.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+const makeSchema = require('../utils/makeSchema');
+
+module.exports = makeSchema({
+  id: '/FieldMetaSchema',
+  type: 'object',
+  description:
+    'Allows for additional metadata to be stored on the field. Only string, numeric or boolean values are allowed',
+  additionalProperties: {
+    type: ['string', 'number', 'boolean'],
```

Comment:
Great question; this is part of the [JSONSchema spec](https://json-schema.org/understanding-json-schema/reference/object#additionalproperties) and allows us to enforce typing on object values. We can't use `properties` to constrain the keys of this `meta` object since a builder should be able to define keys as they like, but we can use `additionalProperties` to make sure the objects types are limited (and can't be other objects, for example).

Specifically, check out this section of the docs linked above: 
![Screen Shot 2024-11-20 at 5 51 08 PM](https://github.com/user-attachments/assets/3ca265cf-1fbf-4207-9dcc-ea5f89738a64)

Reply:
Thanks